geode-core/src/test/java/org/apache/geode/internal/cache/PartitionedRegionTest.java

```diff
@@ -439,6 +443,66 @@ public void generatePRIdShouldNotThrowNumberFormatExceptionIfAnErrorOccursWhileR
         .doesNotThrowAnyException();
   }
 
+  @Test
+  public void getDataRegionForWriteThrowsTransactionExceptionIfNotDataStore() {
+    PartitionedRegion spyPartitionedRegion = spy(partitionedRegion);
+
+    KeyInfo keyInfo = mock(KeyInfo.class);
+    when(keyInfo.getBucketId()).thenReturn(1);
+    doReturn(null).when(spyPartitionedRegion).getDataStore();
+
+    Throwable caughtException =
+        catchThrowable(() -> spyPartitionedRegion.getDataRegionForWrite(keyInfo));
+
+    assertThat(caughtException).isInstanceOf(TransactionException.class).hasMessage(
+        "PartitionedRegion Transactions cannot execute on nodes with local max memory zero");
```

Comment:
These assertions within the tests can be further simplified by using something like the following:

```
    assertThatThrownBy(() -> spyPartitionedRegion.getDataRegionForWrite(keyInfo))
        .isInstanceOf(TransactionException.class)
        .hasMessage("PartitionedRegion Transactions cannot execute on nodes with local max memory zero");
```